### DISCUSSION TITLE
Dicom file, area calculation way ?

### DISCUSSION THREAD

**Discussion by omrylcn** [nroduit/Weasis#552] [Q&A]
Hello Weasis Team,

Thank you for developing this great library. I have a question regarding how the area for a selected polygon or any similar region is calculated in DICOM images using Weasis.

My current formula for calculating the area is straightforward: \( \text{total\_pixels} \times \text{pixel\_spacing}^2 \). However, when I use my formula, the results differ from those displayed in Weasis. For instance, in an image with a pixel spacing of approximately 0.78, the calculated areas do not match.

Below is an example image showing the difference:
![test](https://github.com/nroduit/Weasis/assets/24555636/2831bb35-b84c-42a2-b2e4-21ea4e24d957)

Could you please provide some insight into how Weasis calculates the area for selected regions in DICOM images? Is there an additional factor or method used in the calculation that might account for the discrepancy I am observing?

Thank you in advance for your assistance.

**Comment by nroduit** [comment]
Yes, that's one way of calculating. There are 2 approaches with their advantages and disadvantages:

1. For a vector representation, calculations are performed using mathematical functions.
2. For a pixel representation (raster), this is a special approach in algorithms with discrete mathematics. For example, the area is simply the sum of the pixels. The perimeter, on the other hand, is more difficult to calculate to obtain a good approximation.

In Weasis the tools for geometric measurement (see [area shape implementation](https://github.com/nroduit/Weasis/tree/master/weasis-core/src/main/java/org/weasis/core/ui/model/graphic/imp/area)) use a vector representation and you can find the implementation for every shape.

With[ DICOM segmentation](https://weasis.org/en/tutorials/dicom-segmentation), the surface and the volume are computed with pixel/voxel representation.
